pkg/scheduler/framework/plugins/volumebinding/test_utils.go

```diff
@@ -0,0 +1,249 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volumebinding
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pvutil "k8s.io/kubernetes/pkg/controller/volume/persistentvolume/util"
+	"k8s.io/utils/pointer"
+)
+
+type nodeOpt func(node *v1.Node) error
+
+func makeNode(name string, opts ...nodeOpt) *v1.Node {
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				v1.LabelHostname: name,
+			},
+		},
+	}
+
+	for _, o := range opts {
+		_ = o(node)
+	}
+
+	return node
+}
+
+func nodeWithLabel(key, value string) nodeOpt {
+	return func(node *v1.Node) error {
+		if node.ObjectMeta.Labels == nil {
+			node.ObjectMeta.Labels = map[string]string{}
+		}
+		node.ObjectMeta.Labels[key] = value
+		return nil
+	}
+}
+
+func nodeWithLabelHostname() nodeOpt {
+	return func(node *v1.Node) error {
+		// if node.ObjectMeta.Labels == nil {
+		// 	node.ObjectMeta.Labels = map[string]string{}
+		// }
+		node.ObjectMeta.Labels[v1.LabelHostname] = node.ObjectMeta.Name
+		return nil
+	}
+}
+
+type pvOpt func(pv *v1.PersistentVolume) error
+
+func makePV(name, className string, opts ...pvOpt) *v1.PersistentVolume {
+	pv := &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.PersistentVolumeSpec{
+			StorageClassName: className,
+		},
+	}
+
+	for _, o := range opts {
+		_ = o(pv)
+	}
+	return pv
+}
+
+func pvWithNodeAffinity(keyValues map[string][]string) pvOpt {
+	return func(pv *v1.PersistentVolume) error {
+		matchExpressions := make([]v1.NodeSelectorRequirement, 0)
+		for key, values := range keyValues {
+			matchExpressions = append(matchExpressions, v1.NodeSelectorRequirement{
+				Key:      key,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   values,
+			})
+		}
+		pv.Spec.NodeAffinity = &v1.VolumeNodeAffinity{
+			Required: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchExpressions: matchExpressions,
+					},
+				},
+			},
+		}
+		return nil
+	}
+}
+
+func pvWithVersion(version string) pvOpt {
+	return func(pv *v1.PersistentVolume) error {
+		pv.ObjectMeta.ResourceVersion = version
+		return nil
+	}
+}
+
+func pvWithCapacity(capacity resource.Quantity) pvOpt {
+	return func(pv *v1.PersistentVolume) error {
+		pv.Spec.Capacity = v1.ResourceList{
+			v1.ResourceName(v1.ResourceStorage): capacity,
+		}
+		return nil
+	}
+}
+
+func pvWithPhase(phase v1.PersistentVolumePhase) pvOpt {
+	return func(pv *v1.PersistentVolume) error {
+		pv.Status = v1.PersistentVolumeStatus{
+			Phase: phase,
+		}
+		return nil
+	}
+}
+
+type pvcOpt func(pv *v1.PersistentVolumeClaim) error
```

Comment:
Can we remove the returned error? as I don't spot any functions returning an error, also no error handling on the caller side.